CMakeLists.txt

```diff
@@ -44,7 +44,23 @@ include(CMakeDependentOption)
 message(STATUS "pybind11 v${pybind11_VERSION} ${pybind11_VERSION_TYPE}")
 
 # Check if pybind11 is being used directly or via add_subdirectory
-if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
```

Comment:
Fine to add this, if you want, but is it pybind11's job to educate people on how to use CMake?
As long as pybind11 works with in-place builds, I'm not sure if it's necessary to add this code.

Reply:
The problem is it's really easy to accidentally make an inplace build when you didn't mean to. For example, let's say you are changing a define and forget to include the build directory path:

```cmake
cmake -DCMAKE_BUILD_TYPE=Debug
```

Looks like an innocent mistake, but since you are passing something (doesn't matter what), CMake will go ahead and make an in-place build with your setting. (CMake does protect you from a raw "`cmake`" call, but not when you add anything to it).

Reply:
I can change the text - in-place is supported (as it's the best way to recover exactly what used to happen with out-of-source builds), though not recommended and not required.

Reply:
How about: "You are building in-place. If that is not what you intended to do, you can clean the source directory with:"

Reply:
I changed the text, also dropped the suggestion for how to build (though I rather liked that).